app/models/manipulation.rb

```diff
@@ -0,0 +1,31 @@
+class Manipulation < ActiveRecord::Base
+	belongs_to :account
+	belongs_to :batch
+
+	after_create :change_balance
+	before_destroy :undo
+
+	validates :amount, presence: true
+  validates :issue_date, presence: true
+  validates :action, presence: true
+  validates :description, presence: true
+  validates :account, presence: true
+
+	def change_balance
+		self.account.method(self.action).call(self.amount)
```

Comment:
This is too magical. Other developers will be surprised by this kind of coding. What `change_balance` does will change if the value of `action` is changed. This is confusing.